lib/tesla/middleware/compression.ex

```diff
@@ -49,6 +51,26 @@ defmodule Tesla.Middleware.Compression do
     end
   end
 
+  defmacrop brotli_loaded? do
+    if Code.ensure_loaded?(:brotli) do
+      true
+    else
+      quote do
+        Code.ensure_loaded?(:brotli)
+      end
+    end
+  end
+
+  defmacrop ezstd_loaded? do
+    if Code.ensure_loaded?(:ezstd) do
+      true
+    else
+      quote do
+        Code.ensure_loaded?(:ezstd)
+      end
+    end
+  end
```

Comment:
I'm not sure about this. Wouldn't it be better to provide different middlewares for each compression? 🤔

Reply:
@teamon Honestly, the whole thing is broken right now 😭 I copied some code from @wojtekmach to make it work 😄 

At some point I am going to ping you to decide on that front, The tricky bit is that if you go for the middleware route, you must configure the middleware in the same order as you need to decode. It is practically impossible to know without the server return from my perspective.

On the flip side, just remove it and only support fewer lists... I honestly copied the code without much thought for now

![image](https://github.com/elixir-tesla/tesla/assets/4237280/43809b06-bc22-401f-b218-68a979b20f05)

Reply:
I removed the extra encoding for now